jupyterlab/commands.py

```diff
@@ -1157,7 +1166,15 @@ def _populate_staging(self, name=None, version=None, static_url=None,
         # copy known-good yarn.lock if missing
         lock_path = pjoin(staging, 'yarn.lock')
         if not osp.exists(lock_path):
-            shutil.copy(pjoin(HERE, 'staging', 'yarn.lock'), lock_path)
+            lock_template = pjoin(HERE, 'staging', 'yarn.lock')
+            if self.registry != YARN_DEFAULT_REGISTRY:  # Replace on the fly the yarn repository see #3658
```

Comment:
I don't think this code will work as intended? I think this will only allow the registry config to take effect after a clean operation, since that is the only times that the file is missing.

Reply:
If I change it to replace the registry on-the-fly all the time if this is not the default, the building behavior will be different depending if the registry is the default or not. This proposal ensure coherent behavior. In our infrastructure, the registry is not changing; it is just not the default one. So changing it once works (and I think this is the main use case). But if you think it will be better to apply the change all the time if the registry is not the default, I'll do that.

Reply:
I think it either needs to update every time the config value changes, OR it needs to be clearly documented what the procedure is for changing the registry is. It should be as difficult as possible to get the build system into a inconsistent state, so I would probably put my money on the first option, unless that turns out to be a serious drain on performance (I don't think that will currently be the bottleneck for builds).

> In our infrastructure [...]

If we are exposing this config, it should work for everyone 😉

Reply:
I went for the first option 😉